Mv articles from AGH wiki (#438)
AdguardTeam/KnowledgeBaseDNS: docs/adguard-home/running-securely.md

In the future we plan to release Windows builds as MSI installer files that make sure that this is performed automatically.

## Verify releases

We sign the executable files we build so that you can verify they are really created by us and no one else.  Inside an archive file there's a small file with `.sig` extension which contains the signature data. In a hypothetic situation when the binary file inside an archive is replaced by someone, you'll know that it isn't an official release from AdGuard.

### How to verify that the executable file was built by AdGuard?

1. Unpack the AdGuard Home archive file.

1. Import AdGuard Home public key from keyserver.  For **current releases,** run:

    ```sh
    gpg --keyserver 'keys.openpgp.org' --recv-key '28645AC9776EC4C00BCE2AFC0FE641E7235E2EC6'
    ```

    The above command will print something similar to:

    ```none
    gpg: key 0FE641E7235E2EC6: public key "AdGuard <devteam@adguard.com>" imported
    gpg: Total number processed: 1
    gpg:               imported: 1
    ```

> 1. Verify.  On unix:

Reviewer: I'd prefer uppercasing the name.  But capitalizing it is mandatory.

> 1. Verify.  On unix:

Reviewer: Unix and Windows should be put into separate sub-list items.

The principle is the same on Windows: make sure that the AdGuard Home directory, typically `C:\Program Files\AdGuardHome`, and the `AdGuardHome.exe` binary have the permissions that would only allow regular users to read and execute/list them.

In the future we plan to release Windows builds as MSI installer files that make sure that this is performed automatically.

## Verify releases {#verify-releases}

> We sign the executable files we build so that you can verify they are really created by us and no one else.  Inside an archive file there's a small file with `.sig` extension which contains the signature data. In a hypothetic situation when the binary file inside an archive is replaced by someone, you'll know that it isn't an official release from AdGuard.

Reviewer 2: ```suggestion
We sign the executable files that we build so that you can verify that they were created by us and not by anyone else. Inside an archive file, there is a small file with a `.sig` extension that contains the signature data. If someone replaces the binary file inside an archive, you’ll know it isn’t an official release from AdGuard.
```

### How to verify that the executable file was built by AdGuard? {#how-to-verify-executable}

1. Unpack the AdGuard Home archive file.

> 1. Import AdGuard Home public key from keyserver.  For **current releases,** run:

Reviewer 2: ```suggestion
1. Import the AdGuard Home public key from the keyserver. For **current releases,** run:
```

    ```

1. Verify.

    - On UNIX:

        ```sh
        gpg --verify AdGuardHome/AdGuardHome.sig
        ```

    - On Windows (you might need to install PGP):

        ```ps1
        gpg --verify AdGuardHome/AdGuardHome.exe.sig
        ```

    You'll see something like this:

    ```none
    gpg: assuming signed data in 'AdGuardHome/AdGuardHome'
    gpg: Signature made Mon 15 Aug 2022 19:30:55 MSK
    gpg:                using RSA key 28645AC9776EC4C00BCE2AFC0FE641E7235E2EC6
    gpg:                issuer "devteam@adguard.com"
    gpg: Good signature from "AdGuard <devteam@adguard.com>" [ultimate]
    ```

    Check the following:

    - RSA key: must be `28645AC9776EC4C00BCE2AFC0FE641E7235E2EC6`;
    - issuer name: must be `AdGuard`;
    - E-mail address: must be `devteam@adguard.com`;

    There may also be the following warning:

    ```none
    gpg: WARNING: The key's User ID is not certified with a trusted signature!
    gpg:          There is no indication that the signature belongs to the owner.
    Primary key fingerprint: 2864 5AC9 776E C4C0 0BCE  2AFC 0FE6 41E7 235E 2EC6
    ```

### Reproducing AdGuard Home builds {#reproducing-builds}

> AdGuard Home uses [reproducible builds][repr].  See the “`build-release.sh`” section in our [build script][build] documentation.

Reviewer 2: ```suggestion
AdGuard Home uses [reproducible builds][repr]. See the `build-release.sh` section in our [build script documentation][build].
```

[build]: https://github.com/AdguardTeam/AdGuardHome/tree/master/scripts

> [repr]:  https://reproducible-builds.org/

Reviewer 2: ```suggestion
[repr]: https://reproducible-builds.org/
```
